src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/mbscheduler_hwemu.cxx

```diff
@@ -236,22 +236,33 @@ namespace hwemu {
         cu_bitmap.set(cuidx);
     }
 
-
     /*****************************************************
      * xocl_cu member functions
      *****************************************************/
 
     xocl_cu::xocl_cu(HwEmShim* dev)
     {
         this->xdevice = dev;
+        this->error = false;
+        this->idx = 0;
+        this->uid = 0;
+        this->control = 0;
+        this->dataflow = 0;
+        this->base = 0;
+        this->addr = 0;
+        this->polladdr = 0;
+        this->ap_check = 0;
+        this->ctrlreg = 0;
+        this->done_cnt = 0;
+        this->run_cnt = 0;

```

Comment:
Please use initialization: https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rc-initialize
Or better yet, use in-class initializers: https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rc-in-class-initializer